### PR TITLE
Allow passing `multiprocessing.Queue`s to `QueueListener`/`QueueHandler`

### LINKED ISSUE #10168: `QueueListener` and `QueueHandler` in `logging.handlers` cannot take `multiprocessing.Queue` as queue
`queue` parameter on `__init__` methods on both `logging.handlers.QueueHandler` and `logging.handlers.QueueListener` are annotated as `queue.SimpleQueue[Any] | queue.Queue[Any]`.
as recent change (#10104) made `multiprocessing.Queue` not to inherit `queue.Queue`, type checker complains on passing it to these classes.

```python3
import multiprocessing as mp
from logging.handlers import QueueHandler

QueueHandler(mp.Queue())
```

```
$ pyright --version
pyright 1.1.307
$ pyright --typeshed-path typeshed-f42f665fa mpq.py
0 errors, 0 warnings, 0 informations 
$ pyright --typeshed-path typeshed-adddc4f24 mpq.py
mpq.py
  mpq.py:4:14 - error: Argument of type "Queue[Unknown]" cannot be assigned to parameter "queue" of type "SimpleQueue[Any] | Queue[Any]" in function "__init__"
    Type "Queue[Unknown]" cannot be assigned to type "SimpleQueue[Any] | Queue[Any]"
      "Queue[Unknown]" is incompatible with "SimpleQueue[Any]"
      "Queue[Unknown]" is incompatible with "Queue[Any]" (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations 
```

### PR DESCRIPTION
**Pull request by AlexWaygood**
Fixes #10168

### CODE REVIEW

File: stdlib/logging/handlers.pyi
Comment (AlexWaygood) [inline stdlib/logging/handlers.pyi:268]: We could consider making these classes generic. But best left for another PR.

File: stdlib/logging/handlers.pyi
Comment (JelleZijlstra) [inline stdlib/logging/handlers.pyi:255]: We also need `task_done`, it's used in `QueueListener._monitor`.

Also, `QueueListener.dequeue` has comments suggesting it could be overridden to use something other than `.get()`. I think we can ignore that for typing purposes though: if you make a subclass you can provide different types yourself.

Comment (JelleZijlstra) [reply stdlib/logging/handlers.pyi:255]: Good point, I missed that.

Comment (JelleZijlstra) [inline stdlib/logging/handlers.pyi:255]:
```suggestion
    def put_nowait(self, __item: _T) -> None: ...
```
Pyright is (rightly) being pedantic about the name of the parameter

**Review by JelleZijlstra — APPROVED** [review]
Looks good but there are some random CI failures. Let's wait for GH to get back on its feet and retrigger CI before merging.

**Pull request merged**
